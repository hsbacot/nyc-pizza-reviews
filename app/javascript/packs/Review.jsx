import React from 'react'
import PropTypes from 'prop-types'

export const Review = ({ copy, rating }) => (
  <div>
    <p>{copy}</p>
    <p>User Rating: {rating}</p>
  </div>
)

export default Review
