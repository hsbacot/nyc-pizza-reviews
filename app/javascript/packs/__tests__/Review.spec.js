import React from 'react'
import { Review } from '../Review'
// commented out due to webpack issues with enzyme dependencies and react updates
import { shallow } from 'enzyme'

describe('Review', () => {
  it('should render', () => {
    const component = shallow(<Review />);
    expect(component).toMatchSnapshot();
  })

  it('should handle copy and rating props', () => {
    const component = shallow(<Review copy="test review" rating={4} />);
    expect(component).toMatchSnapshot();
  })
})
