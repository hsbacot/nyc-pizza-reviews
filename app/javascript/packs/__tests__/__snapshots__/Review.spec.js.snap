// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should handle copy and rating props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <p>
        test review
    </p>
    <p>
        User Rating: 
        4
    </p>
</div>,
  "nodes": Array [
    <div>
      <p>
            test review
      </p>
      <p>
            User Rating: 
            4
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Review
        copy="test review"
        rating={4}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "copy": "test review",
          "rating": 4,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <p>
                    test review
          </p>
          <p>
                    User Rating: 
                    4
          </p>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <p>
                    test review
          </p>
          <p>
                    User Rating: 
                    4
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Review
    copy="test review"
    rating={4}
/>,
}
`;

exports[`Review should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <p />
    <p>
        User Rating: 
    </p>
</div>,
  "nodes": Array [
    <div>
      <p />
      <p>
            User Rating: 
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Review />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <p />
          <p>
                    User Rating: 
          </p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <p />
          <p>
                    User Rating: 
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Review />,
}
`;
